pages/price-feeds/use-real-time-data/evm.mdx

```diff
@@ -70,21 +48,38 @@ import "@pythnetwork/pyth-sdk-solidity/PythStructs.sol";
 contract SomeContract {
   IPyth pyth;
 
+  /**
+   * THIS IS AN EXAMPLE CONTRACT. IT IS NOT INTENDED TO BE USED IN PRODUCTION.
+   * THIS CONTRACT IS FOR DEMONSTRATION PURPOSES ONLY.
+   */
+
+  /**
+      * Instantiate IPyth interface with the address of the Pyth contract.
+      * Pyth contract address can be found at https://docs.pyth.network/price-feeds/contract-addresses/evm.
+
+      * @param pythContract The address of the Pyth contract
+      */
   constructor(address pythContract) {
     pyth = IPyth(pythContract);
   }
 
-  function doSomething(
-    uint someArg,
-    string memory otherArg,
-    bytes[] calldata priceUpdateData
-  ) public payable {
-    // Update the on-chain Pyth price(s)
+  /**
+     * This function is an example of how to interact with the Pyth contract.
+
+     * Fetch the updateData from an off-chain service and pass it to the Pyth contract to update the price feeds.
+     * See https://docs.pyth.network/price-feeds/how-pyth-works/hermes for more information on how to fetch the updateData.
+
+     * @param priceUpdateData The updateData to update the contract with the latest price
+     */
+  function fetchPrice(bytes[] calldata priceUpdateData) public payable {
+    // IMPORTANT: Here we fetch the fee required to update the price
+    // and update the contract with the latest price
     uint fee = pyth.getUpdateFee(priceUpdateData);
     pyth.updatePriceFeeds{ value: fee }(priceUpdateData);
 
     // Read the current price
-    bytes32 priceId = 0xf9c0172ba10dfa4d19088d94f5bf61d3b54d5bd7483a322a982e1373ee8ea31b;
+    // The priceIds can be found at https://pyth.network/developers/price-feed-ids
```

Comment:
the reader doesn't know what a priceId is, and this isn't going to help them. You need to introduce the concept and point them to the reference

"Read the current price from a price feed. Each price feed (e.g., ETH/USD) is identified by a price feed id. The complete list of feed ids is available at <link>"